### DISCUSSION TITLE
Bard (PALM) API

### DISCUSSION THREAD

**Discussion by shakeel143** [dsdanielpark/Bard-API#101] [Ideas]
I got the palm api for
hope it helps
https://makersuite.google.com/app/library

**Comment by dsdanielpark** [comment]
Google Bard is a comprehensive service linked to several generative ai models, just like the OpenAI ChatGPT front-end page. Therefore, there may be a backend architecture that performs some functions that generative ai cannot (search, return images, etc.) and connects an appropriate generative ai model.

Google's comprehensive service Bard (currently being provided in the experimental stage) rather than Google's generative ai api. I think the function would be a bit more practical.

Please let me know if there are any further updates.

Also, I will update the contents in scripts and write them in detail again in readme.md.

thank you


## Palm API

Check some test at https://makersuite.google.com/app/prompts/new_text
```
who are you? I am powered by PaLM 2, which stands for Pathways Language Model 2, a large language model from Google AI.
```

1. Official Library https://makersuite.google.com/app/library
2. Get API - https://makersuite.google.com/app/apikey
3. Quick Start Tutorial - https://developers.generativeai.google/tutorials/text_quickstart

```
!pip install -q google-generativeai
```

```python
import pprint
import google.generativeai as palm

palm.configure(api_key='YOUR_API_KEY')

models = [m for m in palm.list_models() if 'generateText' in m.supported_generation_methods]
model = models[0].name
print(model)


prompt = "Who are you????"

completion = palm.generate_text(
    model=model,
    prompt=prompt,
    temperature=0,
    # The maximum length of the response
    max_output_tokens=800,
)

print(completion.result)

**Reply by shakeel143** [reply]
Great explaination, it really help. Thanks.
